packages/module/src/CullingInfo/CullingInfo.tsx

```diff
@@ -0,0 +1,130 @@
+
+const useStyles = createUseStyles({
+  inventoryCullingWarning: {
+    color: 'var(--pf-t--global--icon--color--status--warning--default)',
+  },
+  inventoryCullingDanger: {
+    color: 'var(--pf-t--global--icon--color--status--danger--default)',
+  },
+  iconMargin: {
+    marginRight: 'var(--pf-t--global--spacer--sm)'
+  },
+  messageFont: {
+    fontWeight: 'var(--pf-t--global--font--weight--200)',
+  },
+});
+
+/** extends TooltipProps */
+export interface CullingInformation extends Omit<TooltipProps, 'content'> {
+  /** Option to add custom css classes */
+  className?: string;
+  /** Warning date for when object becomes stale */
+  staleWarning: CullingDate;
+  /** Date when object becomes culled */
+  culled: CullingDate;
+  /** Date when object becomes stale */
+  stale: CullingDate;
+  /** Current date */
+  currDate: CullingDate;
+  /** Optional prop to add custom children */
+  children?: React.ReactElement<any, string | React.JSXElementConstructor<any>> | undefined;
+  /** Option to add custom message ReactElement */
+  render?: Render;
+  /** Optional custom warning message */
+  message?: string;
+}
+
+const CullingInformation: React.FunctionComponent<CullingInformation> = ({
+  culled = new Date(0),
+  className,
+  staleWarning = new Date(0),
+  stale = new Date(0),
+  currDate = new Date(0),
+  children,
+  render,
+  message,
+  ...props
+}) => {
+  const classes = useStyles();
+
+  const calculateTooltip: CalculateTooltip = (culled, warning, currDate) => {
+    const culledDate: Date = new Date(culled);
+    const warningDate: Date = new Date(warning);
+    const diffTime: number = new Date(currDate).valueOf() - warningDate.valueOf();
+    const removeIn: number = Math.ceil((culledDate.valueOf() - new Date(currDate).valueOf()) / days);
+    const msg = message ? message : `System scheduled for inventory removal in ${removeIn} days`;
+    if (diffTime >= 0) {
+      return {
+        isError: true,
+        msg,
+      };
+    }
+  
+    return {
+      isWarn: true,
+      msg,
+    };
+  };
+
+  if (new Date(currDate).valueOf() - new Date(stale).valueOf() < 0) {
+    return render
+      ? render({
+        msg: '',
+      })
+      : children || null;
+  }
+
+  const { isWarn, isError, msg }: CullingInfo = calculateTooltip(culled, staleWarning, currDate);
+  if (render) {
+    return (
+      <span
+        className={clsx({ [classes.inventoryCullingWarning]: isWarn, [classes.inventoryCullingDanger]: isError }, className)}
+      >
+        {isWarn && <ExclamationTriangleIcon className={clsx( classes.iconMargin )}/>}
+        {isError && <ExclamationCircleIcon  className={clsx( classes.iconMargin )}/>}
+        <span className={clsx( classes.messageFont )}>
+          {render({ msg })}
+        </span>
+        
+      </span>
+    );
+  }
+
+  return (
+    <React.Fragment>
+      <Tooltip {...props} content={msg} position="bottom">
+        <span
```

Comment:
From [an accessibility perspective, tooltips](https://www.patternfly.org/components/tooltip/accessibility) need to be placed on interactable elements, such as links or buttons in order for them to able to be accessed by keyboard and screen readers.

Could these components be placed in a plain button, rather than a span?
```
<Button variant="plain" aria-label="Action" icon={<TimesIcon />} />
```

Reply:
The button could be anywhere! The key is just that there needs to be an element which can take browser focus and has some sort of affordance which cues the user that there is more information available if they interact with this icon.